src/pybind/mgr/telemetry/module.py

```diff
@@ -187,81 +200,95 @@ def config_notify(self):
         # wake up serve() thread
         self.event.set()
 
-    def load(self):
-        self.last_upload = self.get_store('last_upload', None)
-        if self.last_upload is not None:
-            self.last_upload = int(self.last_upload)
+    def load(self) -> None:
+        last_upload = self.get_store('last_upload', None)
+        if last_upload is None:
+            self.last_upload = None
+        else:
+            self.last_upload = int(last_upload)
 
-        self.report_id = self.get_store('report_id', None)
-        if self.report_id is None:
+        report_id = self.get_store('report_id', None)
+        if report_id is None:
             self.report_id = str(uuid.uuid4())
             self.set_store('report_id', self.report_id)
+        else:
+            self.report_id = report_id
 
-        self.salt = self.get_store('salt', None)
-        if not self.salt:
+        salt = self.get_store('salt', None)
+        if salt is None:
             self.salt = str(uuid.uuid4())
             self.set_store('salt', self.salt)
+        else:
+            self.salt = salt
 
-    def gather_osd_metadata(self, osd_map):
+    def gather_osd_metadata(self,
+                            osd_map: Dict[str, List[Dict[str, int]]]) -> Dict[str, Dict[str, int]]:
         keys = ["osd_objectstore", "rotational"]
         keys += self.metadata_keys
 
-        metadata = dict()
+        metadata: Dict[str, Dict[str, int]] = dict()
         for key in keys:
             metadata[key] = defaultdict(int)
 
         for osd in osd_map['osds']:
-            res = self.get_metadata('osd', str(osd['osd'])).items()
+            res = self.get_metadata('osd', str(osd['osd']))
             if res is None:
                 self.log.debug('Could not get metadata for osd.%s' % str(osd['osd']))
                 continue
-            for k, v in res:
+            for k, v in res.items():
```

Comment:
This has never worked as intended.

Reply:
it was partly working before the change, except for the case when `res` was `None`...
thanks for fixing this, @tchaikov!